pkg/jobs/registry.go

```diff
@@ -654,6 +654,114 @@ func (r *Registry) LoadJobWithTxn(
 	return j, nil
 }
 
+// GetJobInfo fetches the latest info record for the given job and infoKey.
+func (r *Registry) GetJobInfo(
+	ctx context.Context, jobID jobspb.JobID, infoKey []byte, txn *kv.Txn,
+) ([]byte, bool, error) {
+	row, err := r.ex.QueryRowEx(
+		ctx, "job-info-get", txn,
+		sessiondata.InternalExecutorOverride{User: username.NodeUserName()},
+		"SELECT value FROM system.job_info WHERE job_id = $1 AND info_key = $2 ORDER BY written DESC LIMIT 1",
+		jobID, infoKey,
+	)
+
+	if err != nil {
+		return nil, false, err
+	}
+
+	if row == nil {
+		return nil, false, nil
+	}
+
+	value, ok := row[0].(*tree.DBytes)
+	if !ok {
+		return nil, false, errors.AssertionFailedf("job info: expected value to be DBytes (was %T)", row[0])
+	}
+
+	return []byte(*value), true, nil
+}
+
+// IterateJobInfo iterates though the info records for a given job and info key
+// prefix.
+// TODO(dt): use pagination in query.
```

Comment:
What do you mean by this? The query is streamed. Do you mean, allow the user to resume if they want to paginate?